tests/network/config_services.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test preparing services to use with multimachine scenarios.
+#  At least confiugre:
+#   - http - A basic http support with apache2
+#   - ldap - A simple openladp with users.
+#   - ftp  - simple ftp server
+#  Use on test variables settings: SUPPORT_SERVER_ROLES=http,ldap
+#  The firewall was disable and network configuration used was provide by  tests/network/setup_multimachine.pm
+#  This script was based from tests/support_server/setup.pm, but not use serial
+#  console or neddles. This is to facilites to run on all SLES/OPensuse versions.
+#
+# Maintainer: Marcelo Martins <mmartins@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use mm_network 'setup_static_mm_network';
+use utils qw(systemctl zypper_call exec_and_insert_password);
+
+my $http_server_set = 0;
+my $ldap_server_set = 0;
+my $ftp_server_set  = 0;
+my $setup_script;
+
+sub setup_http_server {
+    #install and configure simple apache2 if $http_server_set;
+    zypper_call('in  apache2');
+    assert_script_run "systemctl stop apache2";
+    assert_script_run 'curl -f -v ' . autoinst_url . "/data/supportserver/http/apache2  >/etc/sysconfig/apache";
+    assert_script_run "systemctl start apache2";
+    $http_server_set = 1;
+}
+
+sub setup_ldap_server {
+    #install and configure basic openldap2 if $ldap_server_set
+    zypper_call('in openldap2');
+    assert_script_run 'systemctl stop slapd';
+    assert_script_run 'curl -f -v ' . autoinst_url . "/data/ldap/slapd.conf > /etc/openldap/slapd.conf";
+    assert_script_run 'curl -f -v ' . autoinst_url . "/data/ldap/test.ldif > /etc/openldap/test.ldif";
+    assert_script_run 'sudo -u ldap slapadd -l /etc/openldap/test.ldif';
+    assert_script_run 'systemctl start slapd';
+    $ldap_server_set = 1;
+}
+
+sub setup_ftp_server {
+    #install and start default ftp server
+    zypper_call('in vsftpd');
+    assert_script_run 'systemctl start vsftpd';
+    $ftp_server_set = 1;
+}
+
+sub run {
+    my ($self) = @_;
+    my $hostname = get_var('HOSTNAME');
+    # Get variable SUPPORT_SERVER_ROLES from job settings.
+    my @server_roles = split(',|;', lc(get_var("SUPPORT_SERVER_ROLES")));
+    my %server_roles = map { $_ => 1 } @server_roles;
+
+    select_console 'root-console';
```

Comment:
As your tests are mainly in textmode, you could try `$self->select_serial_terminal();` instead of `select_console`.
It could speed up your tests a little.

Reply:
ok, done! Thanks